tests/integration/testlib/vxlan.py

```diff
@@ -0,0 +1,84 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import libnmstate
+
+from libnmstate.schema import VXLAN
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceState
+
+VXLAN_IFNAME = '{}.{}'
+
+
+def vxlan_desired_state(vxlans):
+    vxlans_up = []
+    for vxlan in vxlans:
+        vxlans_up.append(vxlan_up(*vxlan))
+    return {Interface.KEY: vxlans_up}
+
+
+@contextmanager
+def vxlan_interfaces(vxlans, extra_interfaces=[]):
+    vxlans_up = []
+    vxlans_absent = []
+    for vxlan in vxlans:
+        vxlans_up.append(vxlan_up(*vxlan))
+        vxlan_id, parent, _ = vxlan
+        vxlans_absent.append(vxlan_absent(vxlan_id, parent))
+    desired_state = {Interface.KEY: vxlans_up + extra_interfaces}
+    try:
+        libnmstate.apply(desired_state)
+        yield desired_state
+    finally:
+        libnmstate.apply({Interface.KEY: vxlans_absent}, verify_change=False)
+
+
+def vxlan_up(vxlan_id, base_if, remote):
+    vxlan_name = vxlan_ifname(vxlan_id, base_if)
+    return {
+        Interface.NAME: vxlan_name,
+        Interface.TYPE: VXLAN.TYPE,
+        Interface.STATE: InterfaceState.UP,
+        VXLAN.CONFIG_SUBTREE: {
+            VXLAN.ID: vxlan_id,
+            VXLAN.BASE_IFACE: base_if,
+            VXLAN.REMOTE: remote,
+            VXLAN.DESTINATION_PORT: 4789,
+        },
+    }
+
+
+def vxlan_absent(vlan_id, base_if):
+    vxlan_name = vxlan_ifname(vlan_id, base_if)
+    return {Interface.NAME: vxlan_name, Interface.STATE: InterfaceState.ABSENT}
+
+
+def vxlan_down(vlan_id, base_if):
+    vxlan_name = vxlan_ifname(vlan_id, base_if)
+    return {
+        Interface.NAME: vxlan_name,
+        Interface.TYPE: VXLAN.TYPE,
```

Comment:
Editing an iface does not require the `type` field.

Reply:
It fails if I don't add type here.

Reply:
You are not supposed to need it once it exists.
Can you check this also for `absent`?